GANDLF/compute/inference_loop.py

```diff
@@ -75,101 +79,96 @@ def inference_loop(inferenceDataFromPickle, device, parameters, outputDir):
-    elif (parameters["modality"] == "path") or (parameters["modality"] == "histo"):
-        # histology inference
-        if os.name != "nt":
-            """
-            path inference is Linux-only because openslide for Windows works only for Python-3.8  whereas pickle5 works only for 3.6 and 3.7
-            """
-            from GANDLF.data.inference_dataloader_histopath import InferTumorSegDataset
-            from openslide import OpenSlide
-
-            # actual computation
-            for _, row in inferenceDataForTorch.iterrows():
-                subject_name = row[parameters["headers"]["subjectIDHeader"]]
-                print(
-                    "Patient Slide       : ",
-                    row[parameters["headers"]["subjectIDHeader"]],
-                )
-                print(
-                    "Patient Location    : ",
-                    row[parameters["headers"]["channelHeaders"]],
-                )
-                print(row[parameters["headers"]["channelHeaders"]].values[0])
-                os_image = OpenSlide(
-                    row[parameters["headers"]["channelHeaders"]].values[0]
-                )
-                level_width, level_height = os_image.level_dimensions[
-                    int(parameters["slide_level"])
-                ]
-                subject_dest_dir = os.path.join(outputDir, subject_name)
-                os.makedirs(subject_dest_dir, exist_ok=True)
-
-                probs_map = np.zeros((level_height, level_width), dtype=np.float16)
-                count_map = np.zeros((level_height, level_width), dtype=np.uint8)
-
-                patient_dataset_obj = InferTumorSegDataset(
-                    row[parameters["headers"]["channelHeaders"]].values[0],
-                    patch_size=patch_size,
-                    stride_size=parameters["stride_size"],
-                    selected_level=parameters["slide_level"],
-                    mask_level=4,
-                )
-
-                dataloader = DataLoader(
-                    patient_dataset_obj,
-                    batch_size=int(parameters["batch_size"]),
-                    shuffle=False,
-                    num_workers=parameters["q_num_workers"],
-                )
-                for image_patches, (x_coords, y_coords) in tqdm(dataloader):
-                    x_coords, y_coords = y_coords.numpy(), x_coords.numpy()
-                    if parameters["model"]["amp"]:
-                        with autocast():
-                            output = model(
-                                image_patches.float().to(parameters["device"])
-                            )
-                    else:
+    elif parameters["modality"] in ["path", "histo"]:
+        # set some defaults
+        if not "slide_level" in parameters:
+            parameters["slide_level"] = 0
+        if not "stride_size" in parameters:
+            parameters["stride_size"] = parameters["patch_size"]
```

Comment:
we like to have a 50% overlap. so maybe patchsize//2?

Reply:
So these are defaults to get the tests to run quicker, you still want to update?

Reply:
Nope. Smart idea.